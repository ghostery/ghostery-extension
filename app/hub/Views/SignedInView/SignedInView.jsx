/**
 * Signed In View
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2019 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import PropTypes from 'prop-types';
import { ExitButton } from '../../../shared-components';

/**
 * A Functional React component for rendering the Login View
 * @return {JSX} JSX for rendering the Login View of the Hub app
 * @memberof HubComponents
 */
const SignedInView = (props) => {
	const signedInAsString = t('hub_signedin_as_email');

	return (
		<div className="SignedInView">
			<ExitButton hrefExit="/" textExit={t('hub_signedin_exit')} />
			<div className="SignedInView--addPaddingTop row align-center">
				<div className="columns small-12 large-10">
					<div className="flex-container align-center-middle">
						<img className="SignedInView__headerImage" src="/app/images/hub/account/ghosty-account.svg" />
						<div className="SignedInView__headerTitle">
							<h3>
								{t('hub_signedin_header_title')}
							</h3>
						</div>
					</div>
				</div>
			</div>
			<div className="SignedInView--addPaddingTop row align-center">
				<div className="columns small-12 medium-6 text-center">
					<h3 className="SignedInView--blueText">
						{`${signedInAsString} ${props.email}`}
					</h3>
				</div>
			</div>
		</div>
	);
};

// PropTypes ensure we pass required props of the correct type
SignedInView.propTypes = {
	email: PropTypes.string.isRequired,
};

export default SignedInView;
