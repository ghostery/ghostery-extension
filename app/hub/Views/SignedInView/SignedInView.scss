/**
 * Signed In View Sass
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

// Signed In View
.SignedInView {
	margin-top: 40px;
	padding-bottom: 40px;
}
.SignedInView--addPaddingTop {
	padding-top: 40px;
}
.SignedInView--blueText {
	color: #00aef0;
}
.SignedInView__headerImage {
	max-width: 180px;
}
.SignedInView__headerTitle h3 {
	font-size: 24px;
	line-height: 1.75;
	color: #4a4a4a;
	margin: 0 25px;
}
