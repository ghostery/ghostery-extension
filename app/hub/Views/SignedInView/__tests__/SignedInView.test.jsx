/**
 * SignedIn View Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2019 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router';
import SignedInView from '../SignedInView';

describe('app/hub/Views/SignedIn component', () => {
	describe('Snapshot tests with react-test-renderer', () => {
		test('signed in view is rendered correctly', () => {
			const initialState = {
				email: 'test@example.com',
			};

			const component = renderer.create(
				<MemoryRouter>
					<SignedInView {...initialState} />
				</MemoryRouter>
			).toJSON();
			expect(component).toMatchSnapshot();
		});
	});

	describe('Shallow snapshot tests rendered with Enzyme', () => {
		test('the happy path of the component', () => {
			const initialState = {
				email: 'test@example.com',
			};

			const component = shallow(<SignedInView {...initialState} />);
			expect(component.find('.SignedInView').length).toBe(1);
			expect(component.find('.SignedInView__headerImage').length).toBe(1);
		});
	});
});
