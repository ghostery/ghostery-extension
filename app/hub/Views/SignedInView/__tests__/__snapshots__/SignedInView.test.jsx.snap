// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/hub/Views/SignedIn component Snapshot tests with react-test-renderer signed in view is rendered correctly 1`] = `
<div
  className="SignedInView"
>
  <a
    aria-current="page"
    className="ExitButton__exit flex-container align-middle active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    <span
      className="ExitButton__exitText"
    >
      hub_signedin_exit
    </span>
    <span
      className="ExitButton__exitIcon"
    />
  </a>
  <div
    className="SignedInView--addPaddingTop row align-center"
  >
    <div
      className="columns small-12 large-10"
    >
      <div
        className="flex-container align-center-middle"
      >
        <img
          className="SignedInView__headerImage"
          src="/app/images/hub/account/ghosty-account.svg"
        />
        <div
          className="SignedInView__headerTitle"
        >
          <h3>
            hub_signedin_header_title
          </h3>
        </div>
      </div>
    </div>
  </div>
  <div
    className="SignedInView--addPaddingTop row align-center"
  >
    <div
      className="columns small-12 medium-6 text-center"
    >
      <h3
        className="SignedInView--blueText"
      >
        hub_signedin_as_email test@example.com
      </h3>
    </div>
  </div>
</div>
`;
